Mathlib/Geometry/Manifold/VectorBundle/LocalFrame.lean

```diff
@@ -283,3 +301,334 @@ end
+
+open scoped Classical in
+-- If x is outside of `e.baseSet`, this returns the junk value 0.
+noncomputable def localFrame
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e]
+    (b : Basis ι 𝕜 F) : ι → (x : M) → V x := fun i x ↦
+  -- idea: take the vector b i and apply the trivialisation e to it.
+  if hx : x ∈ e.baseSet then b.localFrame_toBasis_at e hx i else 0
+
+-- TODO: understand why this isn’t already a simp lemma
+attribute [simp] Trivialization.apply_mk_symm
+
+omit [IsManifold I 0 M] in
+/-- Each local frame `s^i ∈ Γ(E)` of a `C^k` vector bundle, defined by a local trivialisation `e`,
+is `C^k` on `e.baseSet`. -/
+lemma contMDiffOn_localFrame_baseSet
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) (i : ι) :
+    CMDiff[e.baseSet] n (T% (b.localFrame e i)) := by
+  rw [e.contMDiffOn_section_baseSet_iff]
+  apply (contMDiffOn_const (c := b i)).congr
+  intro y hy
+  simp [localFrame, hy, localFrame_toBasis_at]
+
+omit [IsManifold I 0 M] in
+variable (I) in
+/-- `b.localFrame e i` is indeed a local frame on `e.baseSet` -/
+lemma localFrame_isLocalFrameOn_baseSet
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) : IsLocalFrameOn I F n (b.localFrame e) e.baseSet
+    where
+  contMDiffOn i := b.contMDiffOn_localFrame_baseSet _ e i
+  linearIndependent := by
+    intro x hx
+    convert (b.localFrame_toBasis_at e hx).linearIndependent
+    simp [localFrame, hx, localFrame_toBasis_at]
+  generating := by
+    intro x hx
+    convert (b.localFrame_toBasis_at e hx).span_eq.ge
+    simp [localFrame, hx, localFrame_toBasis_at]
+
+omit [IsManifold I 0 M] in
+lemma _root_.contMDiffAt_localFrame_of_mem
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) (i : ι) (hx : x ∈ e.baseSet) :
+    CMDiffAt n (T% (b.localFrame e i)) x :=
+  (b.localFrame_isLocalFrameOn_baseSet I n e).contMDiffAt e.open_baseSet hx _
+
+@[simp]
+lemma localFrame_apply_of_mem_baseSet
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) {i : ι} (hx : x ∈ e.baseSet) :
+    b.localFrame e i x = b.localFrame_toBasis_at e hx i := by
+  simp [localFrame, hx]
+
+@[simp]
+lemma localFrame_apply_of_notMem
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) {i : ι} (hx : x ∉ e.baseSet) :
+    b.localFrame e i x = 0 := by
+  simp [localFrame, hx]
+
+lemma localFrame_toBasis_at_coe
+    (e : Trivialization F (Bundle.TotalSpace.proj : Bundle.TotalSpace F V → M))
+    [MemTrivializationAtlas e] (b : Basis ι 𝕜 F) (i : ι) (hx : x ∈ e.baseSet) :
+    b.localFrame_toBasis_at e hx i = b.localFrame e i x := by simp [hx]
```

Comment:
Do we definitely want this lemma?

It's just `(b.localFrame_apply_of_mem_baseSet e hx).symm`.

Reply:
Fair point: let me remove it for now.